R/teal_data_module.R

```diff
@@ -144,10 +143,17 @@ teal_data_module <- function(ui, server, label = "data module", once = TRUE) {
 teal_transform_module <- function(ui = function(id) NULL,
                                   server = function(id, data) data,
                                   label = "transform module",
-                                  datanames = "all") {
+                                  datanames = character(0)) {
```

Comment:
[DISCUSSION] This begs the question, do we really need "all" or would `character(0)`/`NULL` serve the same purpose on `modules`?

Food for thought, not a blocker on this PR

Reply:
Good question. I don't want to start this discussion now. Just sharing my opinion. I think having `datanames = NULL` to hide sidebar panel is a bad api from us. I think we should not handle `module$datanames = NULL` in teal this way but rather module should have an attribute `show_sidebar = TRUE/FALSE`. 

If we change this show/hide sidebar workflow, then we would be able to properly design `datanames` to be:
- character(0) (unspecified) means all
- <specified> means <specified> 

But I don't create an issue for this. I'm not 100% sure if this is a right way to remove "all", I know for sure that NULL should be abandoned.